tests/test_evalscript.py

```diff
@@ -0,0 +1,105 @@
+import json
+import re
+from typing import List, Optional, Tuple
+
+import pytest
+
+from sentinelhub import DataCollection
+from sentinelhub.data_collections_bands import Unit
+from sentinelhub.evalscript import generate_evalscript
+
+
+def parse_evalscript_info(evalscript: str) -> Tuple[List[str], List[str], List[Tuple[str, str, str]]]:
+    input_names_re = re.search(r"bands: (\[.*\])", evalscript)
+    input_units_re = re.search(r"units: (\[.*\])", evalscript)
+    output_spec = re.findall(r'\{ id: "([^\s.]+)", bands: ([^\s.]+), sampleType: "([^\s.]+)" \}', evalscript)
+
+    assert input_names_re is not None, "regex failed for extraction of input names"
+    assert input_units_re is not None, "regex failed for extraction of input units"
+
+    input_names = json.loads(input_names_re.group(1))
+    input_units = json.loads(input_units_re.group(1))
+
+    return input_names, input_units, output_spec
```

Comment:
damn, you'd almost need to test that this works as intended :see_no_evil: 

could you replace some use of this with `assert expected_str in evalscript`?

Reply:
just to be clear, this is not a test, just a utility to extract test which is then later tested throughout the file.

In this case I would prefer to add some generic simple tests which don't rely on this function and check for simple substring presence

Reply:
I guess we could go about it in a different way, this just allowed me to work with objects instead of strings, but I guess if you have string comparison, everything else is more explicit and intuitive as you construct the string to compare.

I'll try it out

Reply:
managed to remove the utility